iceoryx_examples/iceperf/roudi_main_static_config.cpp

```diff
@@ -25,7 +25,12 @@ int main(int argc, char* argv[])
     static constexpr uint32_t ONE_MEGABYTE = 1024U * 1024;
 
     iox::config::CmdLineParserConfigFileOption cmdLineParser;
-    cmdLineParser.parse(argc, argv);
+    auto cmdLineArgs = cmdLineParser.parse(argc, argv);
+    if (cmdLineArgs.has_error() && (cmdLineArgs.get_error() != iox::config::CmdLineParserResult::INFO_OUTPUT_ONLY))
```

Comment:
Is `INFO_OUTPUT_ONLY` an error or success result ?

Reply:
Neither an error nor a success, but something inbetween. This is why I called it `CmdLineParserResult` and not `CmdLineParserError`. We did the same for removing the `optional` inside a `expected` for the new v1.0 API. However, the methods are still called `get_error()`. We might consider renaming them to `get_result()`.

Reply:
Yeah... but then `get_error` can be a lie !

Reply:
Rust uses the same terminology as we currently do.

@elBoberido What's your take on this?

Reply:
both of you are right ... and I have to leave 😁
`get_unexpected` might be a better name is this case, but I'm not quite sure what's in the `expected` proposal.
As mossmaurice pointed out, Rust does it the same way, although it would be more idiomatic to use a `Result<Option<CmdLineArgs>, E>` in this case.
I'm slightly leaning towards being pragmatic and keep it as it is, since we are doing the same with `takeResult` in the `Subscriber`. We could think of renaming the method, though.